Common/include/toolboxes/C1DInterpolation.hpp

```diff
@@ -0,0 +1,167 @@
+    */
+    bool GetPointMatch() const {return Point_Match;}
+};
+
+class CAkimaInterpolation final: public C1DInterpolation{ 
+protected:
+    vector<su2double> x,y,b,c,d;  /*!< \brief local variables for Akima spline cooefficients */
+    int n; /*!< \brief local variable for holding the size of the vector */
+public:
+    
+    /*!
+    * \brief Constructor of the CAkimaInterpolation class.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    CAkimaInterpolation(vector<su2double> &X, vector<su2double> &Data){
+        SetSpline(X,Data);
+    }
+
+    /*!
+    * \brief Destructor of the CAkimaInterpolation class.
+    */
+    ~CAkimaInterpolation(){}
+
+    /*!
+    * \brief for setting the cofficients for the Akima spline.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    void SetSpline(const vector<su2double> &X, const vector<su2double> &Data) override;
+
+    /*!
+    * \brief For evaluating the value of Akima Spline.
+    * \param[in] Point_Interp - the point where interpolation value is required.
+    * \returns the interpolated value.
+    */
+    su2double EvaluateSpline(su2double Point_Interp) override;
+};
+
+class CLinearInterpolation final: public C1DInterpolation{
+    protected:
+    vector<su2double> x,y,dydx; /*!< \brief local variables for linear 'spline' cooefficients */
+    int n; /*!< \brief local variable for holding the size of the vector */
+    public:
+    
+    /*!
+    * \brief Constructor of the CLinearInterpolation class.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    CLinearInterpolation(vector<su2double> &X, vector<su2double> &Data){
+        SetSpline(X,Data);
+    }
+
+    /*!
+    * \brief Destructor of the CInletInterpolation class.
+    */
+    ~CLinearInterpolation(){}
+
+    /*!
+    * \brief for setting the cofficients for Linear 'spline'.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    void SetSpline(const vector<su2double> &X, const vector<su2double> &Data) override;
+
+    /*!
+    * \brief For evaluating the value for Linear 'spline'.
+    * \param[in] Point_Interp - the point where interpolation value is required.
+    * \returns the interpolated value.
+    */
+    su2double EvaluateSpline(su2double Point_Interp) override;
+};
+
+/*!
+* \brief to correct for interpolation type.
+* \param[in] Inlet_Interpolated - the interpolated data after spline evaluation.
+* \param[in] Theta - the angle of the vertex (in xy plane).
+* \param[in] nDim - the dimensions of the case.
+* \param[in] Coord - the coordinates of the vertex.
+* \param[in] nVar_Turb - the number of turbulence variables as defined by turbulence model
+* \param[in] ENUM_INLET_INTERPOLATIONTYPE - enum of the interpolation type to be done
+* \returns the corrected Inlet Interpolated Data.
+*/
+vector<su2double> CorrectedInletValues(vector<su2double> &Inlet_Interpolated, 
```

Comment:
vector<su2double> &Inlet_Interpolated  --> const vector<su2double> &Inlet_Interpolated, if this is purely an input

Reply:
yes that can be done.